File: python/Arrays/day13/rearrange.py
n = int(input("Enter the size of Array: "))
arr = list(map(int, input(f"Enter {n} numbers: ").split()))

# # Brute-force approach
# pos = []
# neg = []

# for num in arr:
#     if num >= 0:
#         pos.append(num)
#     else:
#         neg.append(num)

# i = 0
# for p, nval in zip(pos, neg):
#     arr[2 * i] = p
#     arr[2 * i + 1] = nval
#     i += 1

# print("Rearranged Array:", arr)





# # optimal approach...
ans = [0] * n
pos = 0
neg = 1

for num in arr:
    if num < 0:
        if neg < n:
            ans[neg] = num
            neg += 2
    else:
        if pos < n:
            ans[pos] = num
            pos += 2

print("Rearranged Array with optimal:", ans)

File: java/Arrays/Day13/Rearrange.java
package Day13;
import java.util.*;

public class Rearrange {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter the size of Array: ");
        int n = sc.nextInt();

        System.out.print("Enter " + n + " " + "numbers: ");

        int arr[] = new int[n];
        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }


        // // brute approach...
        ArrayList<Integer> pos = new ArrayList<>();
        ArrayList<Integer> neg = new ArrayList<>();


        for (int num : arr) {
            if (num >= 0) pos.add(num);
            else neg.add(num);
        }

        for (int i = 0; i < pos.size() && i < neg.size(); i++) {
            arr[2 * i] = pos.get(i);
            arr[2 * i + 1] = neg.get(i);
        }

        System.out.println("Rearranged Array: " + Arrays.toString(arr));


        sc.close();
    }
}